From: Adrian Schmutzler <freifunk@adrianschmutzler.de>
Date: Mon, 13 May 2019 22:31:10 +0200
Subject: babeld: Add openwrt revision to internal version

This is containing custom patches which should be reflected in
the executable version.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

diff --git a/babeld/Makefile b/babeld/Makefile
index 71e777d..b8dc206 100644
--- a/babeld/Makefile
+++ b/babeld/Makefile
@@ -44,8 +44,10 @@ define Package/babeld/conffiles
 /etc/config/babeld
 endef
 
+REVISION:=adsc-$(PKG_VERSION)-$(PKG_RELEASE)
+
 MAKE_FLAGS+= \
-	CFLAGS="$(TARGET_CFLAGS)" \
+	CFLAGS="$(TARGET_CFLAGS) -DBABELD_VERSION='\"$(REVISION)\"'" \
 	LDLIBS="" \
 
 define Package/babeld/install
diff --git a/babeld/patches/0003-Add-option-to-support-version.patch b/babeld/patches/0003-Add-option-to-support-version.patch
new file mode 100644
index 0000000..ae22150
--- /dev/null
+++ b/babeld/patches/0003-Add-option-to-support-version.patch
@@ -0,0 +1,20 @@
+From: Adrian Schmutzler <freifunk@adrianschmutzler.de>
+Date: Mon, 13 May 2019 23:17:20 +0200
+Subject: Add option to support version
+
+Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>
+
+diff --git a/generate-version.sh b/generate-version.sh
+index 22dba7d..7cd00f4 100755
+--- a/generate-version.sh
++++ b/generate-version.sh
+@@ -10,4 +10,6 @@ else
+     version="unknown"
+ fi
+ 
++echo "#ifndef BABELD_VERSION"
+ echo "#define BABELD_VERSION \"$version\""
++echo "#endif"
+-- 
+2.11.0
+
-- 
2.11.0

